### PR TITLE
Added rocksdict as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/help-python-c @conda-forge/help-rust I added `libclang` to build and that made it work for win build, but maybe I need to point to it in some path variable in linux and osx build? Any pointers appreciated here

**Comment by xhochy** [comment]
I have seen this requirement before, but there one needed to put `libclang` into `host`.

**Comment by thewchan** [comment]
> I have seen this requirement before, but there one needed to put `libclang` into `host`.

Thanks @xhochy I'll try that

File: recipes/rocksdict/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - LIBCLANG_PATH=$BUILD_PREFIX/lib  # [not win]
+
+requirements:
+  build:
+    - {{ stdlib('c') }}
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('rust') }}
+    - clangdev
+    - clangxx
```
Comment (xhochy) [inline recipes/rocksdict/meta.yaml:26]: Can you instead add a conda_build_config.yaml with
```
c_compiler:
  - clang
cxx_compiler:
  - clangxx
```

If that works, it's probably the more sustainable approach.

```suggestion
```

Comment (thewchan) [reply recipes/rocksdict/meta.yaml:26]: let me try that thanks @xhochy

Comment (thewchan) [reply recipes/rocksdict/meta.yaml]: @xhochy no luck; should I add `clangdev` back to build? Should I leave `conda_build_config` with clang in or out?

Comment (thewchan) [reply recipes/rocksdict/meta.yaml]: Hmmm actually the error is something else 
```
  - nothing provides requested clang_linux-64 13.*
  - nothing provides requested clangxx_linux-64 13.*
```

**Comment by beckermr** [comment]
@conda-forge-admin relint

File: recipes/rocksdict/meta.yaml (hunk 1 above)
Comment (thewchan) [reply recipes/rocksdict/meta.yaml]: https://conda-forge.org/news/2024/04/30/clang-everywhere/ maybe I need to specify compiler version like it says here?

**Comment by beckermr** [comment]
@conda-forge-admin relint

**Comment by thewchan** [comment]
okay, now it's still complaining about not able to find `clang`, so I think we might have to go with the `clangdev` in `build` solution. What do you think @xhochy ?

**Comment by xhochy** [comment]
> okay, now it's still complaining about not able to find `clang`, so I think we might have to go with the `clangdev` in `build` solution. What do you think @xhochy ?

Yes, seems like the way forward. I would have liked to avoid it but it seems that is sadly not as straight-forward as hoped.

**Comment by thewchan** [comment]
@xhochy @conda-forge/staged-recipes @conda-forge/help-python this is ready to be reviewed again!

**Review by xhochy — APPROVED** [review]

**Pull request merged**